api/src/entity/friendship.entity.ts

```diff
@@ -0,0 +1,39 @@
+import {
+    Entity,
+    Column,
+    CreateDateColumn,
+    UpdateDateColumn,
+    PrimaryColumn,
+} from "typeorm";
+
+export enum FriendshipStatus {
+    PENDING = "pending",
+    ACCEPTED = "accepted",
+    DECLINED = "declined",
+    BLOCKED = "blocked",
+}
+
+@Entity("friendship")
+export class FriendshipEntity {
+    @PrimaryColumn()
+    UserId1: number;
+
+    @PrimaryColumn()
+    UserId2: number;
```

Comment:
Aqui, eu acho melhor termos um ID para cada amizade e usarmos uma [relação many-to-many](https://orkhan.gitbook.io/typeorm/docs/many-to-many-relations) (tipo o que eu fiz pra notificação).

Assim, não precisaremos do userRepository dentro do FriendshipService e poderemos usar o próprio recurso de select do typeorm para retornar os usuários que fazem parte da relação de amizade